bolt-sidecar/src/primitives/mod.rs

```diff
@@ -376,29 +376,65 @@ impl FullTransaction {
     }
 
     /// Returns the sender of the transaction, if recovered.
-    pub fn sender(&self) -> Option<Address> {
-        self.sender
+    pub fn sender(&self) -> Address {
+        self.sender.expect("Recovered sender")
```

Comment:
I'm not a huge fan of quiet panics, let's think of an improvement to this in a later PR.
Not blocking.

Reply:
Reverted